langchain/vectorstores/vector_store_from_doc_index.py

```diff
@@ -0,0 +1,189 @@
+from operator import itemgetter
+from typing import Any, Dict, Iterable, List, Optional, Tuple, Type
+
+try:
+    from docarray import BaseDoc
```

Comment:
current convention is to only import optional dependencies within methods as needed (fine to import same thing across multiple methods). and for type hinting purposes you can do 

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
  from docarray import BaseDoc
  from ...
```
and this will let you use classes from optional imports in type classes without linter causing problems